src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -2768,6 +2789,24 @@ MultiRouterPlannableQuery(Query *query, RelationRestrictionContext *restrictionC
 			{
 				return false;
 			}
+
+			/*
+			 * Currently, we don't support tables with replication factor > 1,
+			 * except reference tables with SELECT ... FOR UDPATE queries. It is
+			 * also not supported from MX nodes.
+			 */
+			if (query->hasForUpdate)
+			{
+				uint32 tableReplicationFactor = TableShardReplicationFactor(
+					distributedTableId);
+
+				EnsureCoordinator();
```

Comment:
We only need to error out if there is a reference table, because then the resource locks are ineffective.